Update Inspirations to 1.14

TeamSpen210 opened the pull request.

TeamSpen210 committed:
Switch cauldron fire check to use a block tag

TeamSpen210 committed:
Convert more config file options

TeamSpen210 committed:
Rewrite Util.getStackFromState() to use loot tables for 1.14

TeamSpen210 committed:
Flatten heartbeet model

TeamSpen210 committed:
Remove plural from block/item texture folders

TeamSpen210 committed:
Flatten materials item model into individual items.

TeamSpen210 committed:
Use correct ID names for droppable materials

TeamSpen210 committed:
Update bookshelf recipe files

TeamSpen210 committed:
Flatten bookshelf blockstates

TeamSpen210 committed:
Flatten path models

TeamSpen210 committed:
Move Charged Arrow model JSON and simplify

TeamSpen210 committed:
Remove plural on item folder name in models

TeamSpen210 committed:
Remove plural from block folder name in models

TeamSpen210 committed:
Convert all language files to JSON

TeamSpen210 committed:
Flatten out all language keys to individual IDs

TeamSpen210 committed:
Remove plurals in folders in blockstates

TeamSpen210 committed:
Flatten enlightened bush blockstate definition

TeamSpen210 committed:
Flatten Enlightened Bush blockstate files

TeamSpen210 committed:
Fix location of wooden trapdoor tag

TeamSpen210 committed:
Update flower recipes to 1.14 format

TeamSpen210 committed:
Update path block recipes for 1.14

TeamSpen210 committed:
Update rope recipes for 1.14

TeamSpen210 committed:
Update colored book recipes to 1.14

TeamSpen210 committed:
Add a bunch of generic item tags for crafting

TeamSpen210 committed:
Add tag for Waypoint Compasses

TeamSpen210 committed:
Update lots of recipes to 1.14

TeamSpen210 committed:
Use the correct name for the smooth stone texture

TeamSpen210 committed:
Flatten Mulch blockstate files

TeamSpen210 committed:
Fix typo

TeamSpen210 committed:
Document the special functionality of the cauldron_fire tag.

TeamSpen210 committed:
Add copies of block tags for their item versions

TeamSpen210 committed:
Update mulch recipes for 1.14

TeamSpen210 committed:
Add vines tag

TeamSpen210 committed:
Fix boolean value type

TeamSpen210 committed:
Add more inspirations-specific tags

TeamSpen210 committed:
Update Redstone Barrel for 1.14

TeamSpen210 committed:
Fix the remaining Bookshelf methods for 1.14

TeamSpen210 committed:
Add more item tags to replace config options

TeamSpen210 committed:
Update most of the 1.14 config

TeamSpen210 committed:
Partially update InspirationsRegistry

TeamSpen210 committed:
Update Carpeted Pressure Plates for 1.14

TeamSpen210 committed:
Remove InspirationsOredict

This is replaced by tags in the datapack.

TeamSpen210 committed:
Always register items, but hide if disabled.
This is required by 1.14 configs.

TeamSpen210 committed:
Fix a bunch of syntax errors in datapack

TeamSpen210 committed:
Implement library.ClientUtil.onResourceReload

TeamSpen210 committed:
Update networking code for 1.14

TeamSpen210 committed:
Make Building pulse run

TeamSpen210 committed:
Fix a lot of mistakes in the resource packs

TeamSpen210 committed:
Increment versions and fix access transformer

TeamSpen210 committed:
Switch glass door to non-forge blockstates

TeamSpen210 committed:
Update crop blocks for 1.14

TeamSpen210 committed:
Update RecipeUtil for post-Flattening

TeamSpen210 committed:
Strip unusable stateMappers

TeamSpen210 committed:
Use unique collisions for each path type

TeamSpen210 committed:
Make rope item quantity math work correctly.

TeamSpen210 committed:
Add tag for milk containers

TeamSpen210 committed:
Reorganise main Mod for 1.14
Most modules commented out until fully functional

TeamSpen210 committed:
Update fitted carpet blockstates/models for 1.14 names

TeamSpen210 committed:
Reimplement fitted carpets for 1.14

To comply with the new blockstate restrictions for vanilla overrides,
use two blocks for each color.

TeamSpen210 committed:
Make flowers non-solid

TeamSpen210 committed:
Support waterlogging ropes

TeamSpen210 committed:
Don't double-name blocks/items we register

This isn't permitted.

TeamSpen210 committed:
Use the correct names for small flowers, fixing the models

TeamSpen210 committed:
Fix appearance of coloured books

TeamSpen210 committed:
Register all client proxies with both event busses correctly

TeamSpen210 committed:
Fix duplicate config names

TeamSpen210 committed:
Fix cyan carpet models

TeamSpen210 committed:
Fix a bunch of recipes

TeamSpen210 committed:
Fully implement custom crop blocks

TeamSpen210 committed:
Fix model for mulch blocks

TeamSpen210 committed:
Fix model for vines and ropes

TeamSpen210 committed:
Fix model for glass trapdoor/door

TeamSpen210 committed:
Implement correct drops for fitted carpet blocks

TeamSpen210 committed:
Specify drops for carpeted trapdoors

TeamSpen210 committed:
Implement Trapped Book recipe

TeamSpen210 committed:
Port over Tweaks module events

TeamSpen210 committed:
Make main mod class use the Mod event bus properly.

TeamSpen210 committed:
Clear out old client model registering code

TeamSpen210 committed:
Reimplent Bookshelf GUI for 1.14

TeamSpen210 committed:
Add Brewing recipe variant which is disableable

TeamSpen210 committed:
Add a flag to recognize when the config is ready to be read

TeamSpen210 committed:
Add helper method for registering entities

TeamSpen210 committed:
Fix this config option

TeamSpen210 committed:
Update Tweaks enchantments for 1.14

TeamSpen210 committed:
Update Redstone Arrow / Charged Arrow for 1.14

TeamSpen210 committed:
Partially rewrite ToolsEvents for 1.14

Most of these events are currently broken in Forge...

TeamSpen210 committed:
Fix names in waypoint compass dying recipes

TeamSpen210 committed:
Fix various datapack errors

TeamSpen210 committed:
Update item property getters for 1.14

TeamSpen210 committed:
Update Waypoint Compass for 1.14

TeamSpen210 committed:
Partially update Crooks for 1.14

Requires event updates in Forge before they function.

TeamSpen210 committed:
Reimplement Waypoint Compass tinting handler

TeamSpen210 committed:
Register the potato/carrot seeds and carpets

TeamSpen210 committed:
Make the LockableTileEntity fields public

TeamSpen210 committed:
Add ToolTypes for various tools

TeamSpen210 committed:
Enable Tools module

TeamSpen210 committed:
TileEntityBeacon.isComplete -> BeaconTileEntity.beamSegments in 1.14

TeamSpen210 committed:
Switch Util.clickedAABB to take a vector

The only caller has values packed into one already.

TeamSpen210 committed:
Update Torch Lever to 1.14

Split the block into two block implementations like Vanilla has done.
The two share very little code anyway.

TeamSpen210 committed:
Enable Utility module

TeamSpen210 committed:
Remove old 1.12 Utilities statemappers

TeamSpen210 committed:
Fully implement Redstone Barrels for 1.14

TeamSpen210 committed:
Update carpeted pressure plates to 1.14

TeamSpen210 committed:
Update carpeted trapdoors to 1.14

TeamSpen210 committed:
Fix Torch Levers being placed in an activated state

TeamSpen210 committed:
Add missing model file

TeamSpen210 committed:
Update Bricks Button to 1.14
- The nether bricks model needs to be adjusted to match 1.14's new textures.

TeamSpen210 committed:
Tweak holding poses for a few items

TeamSpen210 committed:
Update Pipe to 1.14

TeamSpen210 committed:
Update Collector to 1.14

TeamSpen210 committed:
Register the bricks button items and blocks

TeamSpen210 committed:
Tint sugar cane crop to match the biome

TeamSpen210 committed:
Make North Compass rotate on floor/ceiling item frames

TeamSpen210 committed:
Properly clear caches when reloading resources

TeamSpen210 committed:
Update forge to the latest version

This has the player events reimplemented.

TeamSpen210 committed:
Detect beacon completion status correctly for 1.14

TeamSpen210 committed:
Use the translated tooltip text for waypoint compass in advanced mode

TeamSpen210 committed:
Add missing loot tables for lots of blocks.

TeamSpen210 committed:
Fix a bunch of recipe errors

TeamSpen210 committed:
Make carpeted pressure plates use carpet strength and tool types

TeamSpen210 committed:
Use the expected names for the additional potion items

TeamSpen210 committed:
Don't have the pipe item hover low in item form

TeamSpen210 committed:
Update and disable tank fluid behaviour

TeamSpen210 committed:
Update logic for placing carpets on pressure plates

TeamSpen210 committed:
Make hanging vine dropping behavior function properly

TeamSpen210 committed:
Update fireworks and enchanted book tint handlers for 1.14

TeamSpen210 committed:
Register Collector and Pipe GUIs

TeamSpen210 committed:
Don't have bookshelves try and place vertically

TeamSpen210 committed:
Update maven identifiers for 1.14

TeamSpen210 committed:
Use 1.14.4 version of JEI

TeamSpen210 committed:
Make cauldron particle constructor callable

TeamSpen210 committed:
Update TOP version

TeamSpen210 committed:
Apply 1.14 renames in plugins

TeamSpen210 committed:
Update dyed water bottles to 1.14

TeamSpen210 committed:
Shift position of Nether Bricks Button to match 1.14 textures

TeamSpen210 committed:
Make Waypoint Compass copying recipe disableable

TeamSpen210 committed:
Drop inventory contents when blocks are destroyed.

TeamSpen210 committed:
Precompute pipe voxelshapes

TeamSpen210 committed:
Fix pipe upwards placement

TeamSpen210 committed:
Document these blockstates

TeamSpen210 committed:
Reimplement 8bcdcc7ae4ec3ecc: Allow ropes to attach to fence posts

TeamSpen210 committed:
Reimplement 3d5c6f187b0fd221: Add rope ladders: right click a rope with sticks to add rungs

TeamSpen210 committed:
Remove Redstone Torch Lever

TeamSpen210 committed:
Remove Extended Flower Pot

TeamSpen210 committed:
Add rope ladder config option

TeamSpen210 committed:
Move vanilla replacement configs to separate file, read early on.

TeamSpen210 committed:
Convert over Enchantable Shield class

TeamSpen210 committed:
Partially convert TexturedModel logic to 1.14
This needs the Forge blockstate fix to visibly function.

TeamSpen210 committed:
Re-enable isXLoaded() code in PulseBase

TeamSpen210 committed:
Fix imports here

TeamSpen210 committed:
This function doesn't need MC to be passed in

TeamSpen210 committed:
Fix path and rope collision shapes

TeamSpen210 committed:
Remove flower registry code

TeamSpen210 committed:
Reformat

TeamSpen210 committed:
This is now a function

TeamSpen210 committed:
Remove more flower pot bits

TeamSpen210 committed:
Update Anvil-dispensing behaviour for 1.14

TeamSpen210 committed:
Remove the "missing" potion types.

TeamSpen210 committed:
Reimplement 5f22c9ca3831f09c: Improve waypoint compass tooltip

Config option now also responds to reducedDebugInfo
Now shows dimension ID in the advanced tooltip, in case one type is reused
Dimension names are now cleaner when not localized, localized a few other dimension mods

TeamSpen210 committed:
Reimplement 4b2e32cfd836eb2c:  Switch from instanceof ItemShield to Item::isShield for enchantment checks

TeamSpen210 committed:
Reimplement 4f0cca71d6c243ff: Instead of grass always winning ties, randomly choose grass or mycelium

TeamSpen210 committed:
Reimplement e512589a8779470c: Add a line break between shield patterns and the enchantments

TeamSpen210 committed:
Implement colorised Enchanted Books and Fireworks for 1.14

TeamSpen210 committed:
Reimplement adding custom portal colors for 1.14

TeamSpen210 committed:
Use the correct config functions, now brewing potions is removed

TeamSpen210 committed:
Reformat ClientUtil

TeamSpen210 committed:
Reformat Glass Trapdoor block

TeamSpen210 committed:
Fix collision shape for ropes

TeamSpen210 committed:
Update milk cooldown logic for 1.14

TeamSpen210 committed:
Reimplement 276b59df8f9b3d50: Improve carpet stair hitboxes

TeamSpen210 committed:
Associate fitted carpets with the vanilla item as well.

TeamSpen210 committed:
Use the correct new name for the access transformer file

TeamSpen210 committed:
Reimplement 62a424319cfc3ff9: Skeletons now drop skulls and cave spiders drop webs

TeamSpen210 committed:
Remove silverfish/wither bone loot tables

These don't function right now.

KnightMiner commented on build.gradle:
```diff
@@ -36,10 +37,62 @@ if (System.getenv().BUILD_NUMBER != null) {
-    runDir = "run"
-    replace '${version}', project.version
+	mappings channel: 'snapshot', version: config.mappings_version
+//    replace '${version}', project.version
```
This is not needed, it was just to sub the version into the main class file which should be in the JAR manifest now, see below

KnightMiner commented on build.gradle:
```diff
@@ -66,45 +119,51 @@ repositories {
+//    from(sourceSets.main.resources.srcDirs) {
+//        exclude 'mcmod.info'
+//    }
+//}
```
This is not needed, it was just to sub the version into mcmod.info which should be in the JAR manifest now, see below

KnightMiner commented on build.gradle:
```diff
@@ -66,45 +119,51 @@ repositories {
 jar {
     manifest {
-        attributes 'FMLAT': 'inspirations_at.cfg'
+        attributes 'FMLAT': 'accesstransformer.cfg'
```
I don't know that we need this here still, I have an AT in Simply Tea without it.

You should also add attributes to this, see [what I did in Simply Tea](https://github.com/elucent/SimplyTea/blob/1.14/build.gradle#L84-L92)

KnightMiner commented on build.properties:
```diff
@@ -1,10 +1,12 @@
+mod_version=0.3.0
+minecraft_version=1.14.4
+minecraft_base_version=1.14
+minecraft_version_jei=1.14.4
```
can you not use minecraft_version for this?

KnightMiner commented on build.properties:
```diff
@@ -1,10 +1,12 @@
 hwyla_version=1.8.23-B38_1.12
-probe_version=1.12-1.4.12-6
\ No newline at end of file
+probe_version=1.14-1.4.30-6
```
Not sure why I made this 1.14-version when I split the rest... can probably remove the 1.14 and make build.gradle do {base-minecraft-version}-{probe-version}

KnightMiner commented on src/main/java/knightminer/inspirations/Inspirations.java:
```diff
@@ -1,83 +1,124 @@
+@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD)
 public class Inspirations {
 	public static final String modID = "inspirations";
 	public static final String modVersion = "${version}";
```
I don't think this is used anywhere, can delete as its in mods.toml now

KnightMiner commented on src/main/java/knightminer/inspirations/Inspirations.java:
```diff
@@ -1,83 +1,124 @@
-	public static GuiHandler guiHandler = new GuiHandler();
+		MinecraftForge.EVENT_BUS.register(pulseManager);
+		MinecraftForge.EVENT_BUS.addListener(this::registerRecipeTypes);
+		FMLJavaModLoadingContext.get().getModEventBus().register(this);
```
Why do you do this and annotate the class? I think the class annotation is probably the one that can be removed as its for static event listeners

TeamSpen210 replied:
For that reason, the annotation wasn't working.

KnightMiner commented on src/main/java/knightminer/inspirations/building/BuildingClientProxy.java:
```diff
@@ -189,50 +138,46 @@ public void registerItemColors(ColorHandlerEvent.Item event) {
 
 			// if the type has it's own color, use that
-			int color = LightsType.fromMeta(stack.getMetadata()).getColor();
+			int color = ((BlockEnlightenedBush)((BlockItem)stack.getItem()).getBlock()).getColor();
```
Not a fan of these unchecked casts, even though its very unlikely to be the wrong class

KnightMiner commented on src/main/java/knightminer/inspirations/building/BuildingEvents.java:
```diff
@@ -2,60 +2,131 @@
 public class BuildingEvents {
+	private static ToolType shearsType = ToolType.get("shears");
 	@SubscribeEvent
 	public static void addFlowerDrop(HarvestDropsEvent event) {
```
I'd like to move this to loot tables if possible, not sure what Forge supports for modifying vanilla tables though. Ideally we would just cancel the vanilla drops if our table succeeds

KnightMiner commented on src/main/java/knightminer/inspirations/building/InspirationsBuilding.java:
```diff
@@ -1,166 +1,228 @@
-	public static BlockPath path;
-	public static Block flower;
-	public static BlockEnlightenedBush enlightenedBush;
+	public static Map<BlockMulch.MulchColor, Block> mulch=new HashMap<>();
```
Not sure how I feel about hash maps for these as I rarely need to fetch a block specific after registering. There is a utility in Mantle that registers a list of blocks from a property, or it might make sense to make a similar method that accepts a string array of variant names.

For most decorative blocks I'd say just register multiple copies of a no parameter constructor. For ones that need tinting (like enlightened bushes), pass in the prop as a parameter (I think the mantle method supports that, might need to modify it).

KnightMiner commented on src/main/java/knightminer/inspirations/building/InspirationsBuilding.java:
```diff
@@ -1,166 +1,228 @@
-	@Subscribe
-	public void init(FMLInitializationEvent event) {
+	@SubscribeEvent
+	public void init(InterModEnqueueEvent event) {
```
I wonder if this should be FMLCommonSetupEvent instead. Not sure about how all the new events go in order

KnightMiner commented on src/main/java/knightminer/inspirations/building/block/BlockBookshelf.java:
```diff
@@ -1,127 +1,114 @@
-import java.util.Locale;
 
-public class BlockBookshelf extends BlockInventory implements ITileEntityProvider {
+public class BlockBookshelf extends InventoryBlock implements ITileEntityProvider {
```
I'd like to use the new naming, so this would be BookshelfBlock. Diddo for all other blocks. Since its a lot of renames it might make the PR harder to read, so might be better to hold off on that until after the merge.

KnightMiner commented on src/main/java/knightminer/inspirations/building/block/BlockBookshelf.java:
```diff
@@ -312,95 +272,64 @@ public boolean canProvidePower(IBlockState state) {
 	@Override
-	public void getDrops(NonNullList<ItemStack> drops, IBlockAccess world, BlockPos pos, IBlockState state, int fortune){
-		drops.add(TextureBlockUtil.getBlockItemStack(world, pos, state));
+	public List<ItemStack> getDrops(@Nonnull BlockState state, LootContext.Builder builder) {
```
I'd like to work this into a loot table if possible, depends how much context loot table functions give us

KnightMiner commented on src/main/java/knightminer/inspirations/building/block/BlockBookshelf.java:
```diff
@@ -193,101 +180,74 @@ private static int bookClicked(EnumFacing facing, float clickX, float clickY, fl
+			builder.put(side, VoxelShapes.or(
+					VoxelShapes.create(x1,  0,      z1,  x2,  0.0625, z2), // bottom shelf
+					VoxelShapes.create(x1,  0.4375, z1,  x2,  0.5625, z2), // middle shelf
+					VoxelShapes.create(x1,  0.9375, z1,  x2,  1,      z2), // top shelf
```
This might be cleaner if you just make the half block shape then cut out the shelves using VoxelShape::combineAndSimplify with IBooleanFunction.ONLY_FIRST, like used in ComposterBlock. You can OR together the empty space for the two shelves then subtract it from the full shelf shape

KnightMiner commented on src/main/java/knightminer/inspirations/building/block/BlockEnlightenedBush.java:
```diff
@@ -1,134 +1,122 @@
+
+public class BlockEnlightenedBush extends Block {
+
+	public static final EnumProperty<LightsType> LIGHTS = EnumProperty.create("lights", LightsType.class);
```
Feel free to skip since these are unfinished, but this should be separate blocks instead of a prop now

KnightMiner commented on src/main/java/knightminer/inspirations/building/block/BlockFlower.java:
```diff
@@ -1,157 +1,98 @@
+import javax.annotation.Nullable;
 
-	/* Blockstate */
+public class BlockFlower extends BushBlock implements IGrowable {
```
Can this extend the vanilla flower block now? Only reason I did not before was because it forced me to use the vanilla flower type property.

Extending vanilla lets a lot of this redundant logic be removed

TeamSpen210 replied:
Actually we can't, since the vanilla block requires a potion effect for [Suspicious Stew](https://minecraft.gamepedia.com/Suspicious_Stew), though we probably should/could add that. `getCollisionShape()` is redundant since I forgot `doesNotBlockMovement()`, but everything else is specific for the double-height growing ability.

KnightMiner commented on src/main/java/knightminer/inspirations/building/block/BlockMulch.java:
```diff
@@ -2,89 +2,53 @@
-	@Override
-	public int damageDropped(IBlockState state) {
-		return getMetaFromState(state);
+	public BlockMulch(MulchColor color) {
```
Don't need mulch color in the constructor as its never used, can just be ditched overall

KnightMiner commented on src/main/java/knightminer/inspirations/building/block/BlockMulch.java:
```diff
@@ -2,89 +2,53 @@
-			list.add(new ItemStack(this, 1, type.getMeta()));
-		}
-	}
+	protected static final AxisAlignedBB BOUNDS = new AxisAlignedBB(0.0D, 0.0D, 0.0D, 1.0D, 0.9375D, 1.0D);
```
Shouldn't this be VoxelShape now? This is just the physical hitbox rather than the visible one, so follow soul sand as a pattern.

KnightMiner commented on src/main/java/knightminer/inspirations/building/block/BlockPath.java:
```diff
@@ -1,116 +1,97 @@
 
-	/* Block Shape */
+	private final VoxelShape shape;
+	private final VoxelShape coll_shape;
```
Nitpicky, but I normally use camelcase for variables (with the possible exception of registry entries in the module classes)

KnightMiner commented on src/main/java/knightminer/inspirations/building/block/BlockRope.java:
```diff
@@ -1,176 +1,160 @@
+	private final VoxelShape[] shape;
+	private final VoxelShape[] shape_bottom;
+
+	public BlockRope(Properties props, Item rungsItem, VoxelShape[] shape, VoxelShape[] shape_bottom) {
```
Personally, I would just use the rope bounds in this class and make a chain class that overrides and swaps the bounds, saves needing to expose bounds as public/prevents someone passing an invalid bound amount

KnightMiner commented on src/main/java/knightminer/inspirations/common/Config.java:
```diff
@@ -1,69 +1,64 @@
+
+
+	public static BooleanValue enableBookshelf;
+	private static BooleanValue enableColoredBooksRaw;
```
Minor, but this is named `enableColoredBooksRaw` while for rope ladders you just called it `enableRopeLadder`. I like it without "raw" personally

KnightMiner commented on src/main/java/knightminer/inspirations/common/Config.java:
```diff
@@ -72,81 +67,130 @@
+
+	private enum SpongeEmptyCauldron {
+		FALSE, // No emptying.
+		TRUE, // For any amount of liquid.
```
Before I was kinda pretending it was a boolean, but since we are making this an enum anyways, should switch it to FALSE, ANY, FULL or maybe DISABLE, ANY, FULL

KnightMiner commented on src/main/java/knightminer/inspirations/common/Config.java:
```diff
@@ -789,122 +1002,83 @@ private static void processCauldronRecipes(String[] cauldronRecipes) {
-			String prop = JsonUtils.getString(json, "prop");
+		public BooleanSupplier parse(@Nonnull JsonObject json) {
+			String prop = JSONUtils.getString(json, "prop");
 			return () -> propertyEnabled(prop);
```
Original I set it up this way so I would not have to write `()-> enableFeature` for every case, but given that config props are already boolean suppliers I would move the switch statement in here write the properties like `enableFeature::get`

KnightMiner commented on src/main/java/knightminer/inspirations/common/PulseBase.java:
```diff
@@ -3,39 +3,26 @@
-import slimeknights.mantle.item.ItemBlockMeta;
-import slimeknights.mantle.item.ItemBlockSlab;
 
 public class PulseBase {
```
Should extend Mantle's IRegisterUtil as that contains a lot of the register methods. If anything is missing just implement it here and I'll move it over later.

KnightMiner commented on src/main/java/knightminer/inspirations/common/PulseBase.java:
```diff
@@ -134,61 +92,52 @@ protected static ItemBlockMeta registerItemBlock(IForgeRegistry<Item> registry,
+	 * @param event LootTableLoadEvent
+	 * @param name  Name of vanilla table and the inspirations table
 	 */
 	protected static void addToVanillaLoot(LootTableLoadEvent event, String name) {
```
I have something similar to this working in Simply Tea [here](https://github.com/elucent/SimplyTea/blob/1.14/src/main/java/knightminer/simplytea/core/Events.java#L67-L78), should be able to copy it over (ignore the TODO comment)

TeamSpen210 replied:
Currently this is unused now, when I copied over the spider/skeleton drops I just used `LivingDropsEvent`.

KnightMiner replied:
Ideally I would prefer to do loot tables as that is the way Forge is moving.

KnightMiner commented on src/main/java/knightminer/inspirations/common/PulseBase.java:
```diff
@@ -134,61 +92,52 @@ protected static ItemBlockMeta registerItemBlock(IForgeRegistry<Item> registry,
 
-	protected static void registerDispenserBehavior(Block block, IBehaviorDispenseItem behavior) {
-		if(block != null) {
+	protected static void registerDispenserBehavior(Block block, IDispenseItemBehavior behavior) {
```
Not really needed anymore, it was needed when every block was possibly optionally null (because config) but now everything always registers in its own module.

KnightMiner commented on src/main/java/knightminer/inspirations/common/item/ItemBlockTexture.java:
```diff
@@ -1,26 +1,49 @@
-public class ItemBlockTexture extends ItemBlockMeta {
-	public ItemBlockTexture(Block block) {
-		super(block);
+public class ItemBlockTexture extends BlockItem implements IHidable {
```
Just extend HidableBlockItem, it does the hidable stuff already

KnightMiner commented on src/main/java/knightminer/inspirations/common/network/MilkablePacket.java:
```diff
@@ -1,53 +1,57 @@
+			throw new UnsupportedOperationException("Clientside only");
+		}
+
+		Entity entity = Minecraft.getInstance().world.getEntityByID(entityID);
```
Is this server safe? Since this method is accessible on both sides?

KnightMiner commented on src/main/java/knightminer/inspirations/library/InspirationsRegistry.java:
```diff
@@ -1,31 +1,34 @@
-import net.minecraft.block.BlockMushroom;
-import net.minecraft.block.BlockSapling;
-import net.minecraft.block.BlockTallGrass;
+import net.minecraft.block.*;
```
Don't use wildcard imports, makes the code harder to read without an IDE, which I do more often than you'd expect

KnightMiner commented on src/main/java/knightminer/inspirations/library/InspirationsRegistry.java:
```diff
@@ -41,15 +44,30 @@
+	// Blocks with this tag act as fire for the cauldron.
+	public static final Tag<Block> TAG_CAULDRON_FIRE = new BlockTags.Wrapper(new ResourceLocation(Inspirations.modID, "cauldron_fire"));
+	// Items with this tag are registered to have fluid tank functionality.
+	public static final Tag<Item> TAG_DISP_FLUID_TANKS = new ItemTags.Wrapper(new ResourceLocation(Inspirations.modID, "fluid_containers"));
```
I'm not sure if this loads early enough to work, wouldn't it load too late to add to the dispenser registry? Though its something I probably don't need to deal with until the cauldron rewrite.

TeamSpen210 replied:
Not really sure about that and the rest of the dispenser functionality. Without registering something on every item or futzing with the map whenever the config changes, not sure if you can do anything and be compatible here...

KnightMiner replied:
I want to avoid registering something on every item, but not sure what other choice I have. Might have to hold off on the whole dispenser automation for now

KnightMiner commented on src/main/java/knightminer/inspirations/building/BuildingEvents.java:
```diff
@@ -2,60 +2,131 @@
+import net.minecraftforge.items.ItemHandlerHelper;
 
 public class BuildingEvents {
+	private static ToolType shearsType = ToolType.get("shears");
```
Use the property from InspirationsRegistry

KnightMiner commented on src/main/java/knightminer/inspirations/library/InspirationsRegistry.java:
```diff
@@ -65,299 +83,202 @@ public static void setConfig(String key, boolean value) {
+	 * @deprecated use {@link #registerBook(Item, float)}
 	 */
 	@Deprecated
 	public static void registerBook(ItemStack stack, boolean isBook) {
```
You can remove these deprecated methods, they were just to not make breaking changes before 1.14

KnightMiner commented on src/main/java/knightminer/inspirations/library/InspirationsRegistry.java:
```diff
@@ -41,15 +44,30 @@
 	public static final Logger log = Util.getLogger("api");
 
+	// Blocks with this tag act as fire for the cauldron.
+	public static final Tag<Block> TAG_CAULDRON_FIRE = new BlockTags.Wrapper(new ResourceLocation(Inspirations.modID, "cauldron_fire"));
```
Can use `Util::getResource` for the Inspirations resource locations

TeamSpen210 replied:
I'm unsure, should `library.*` not use anything outside of that for an "API"? If everything here is moved to JSON that might not be required though.

KnightMiner replied:
Anything in library can use anything in library, just should not use anything outside of library. Ideally I want to be able to publish a lib jar in the future. Have not followed that practice the best though.

What probably needs to be done is moving the mod ID constant to the util class.

KnightMiner commented on src/main/java/knightminer/inspirations/library/Util.java:
```diff
@@ -43,21 +50,26 @@ public static ResourceLocation getResource(String res) {
 		// translates twice to allow rerouting/alias
-		return I18n.translateToLocal(I18n.translateToLocal(String.format(key, pars)).trim()).trim();
+		String trans = new TranslationTextComponent(String.format(key, pars)).getUnformattedComponentText().trim();
+		return new TranslationTextComponent(trans).getUnformattedComponentText().trim();
```
Honestly, I really don't need to translate it twice, thats leftover from stealing this method from Tinkers Construct.

Might be able to delete this entirely if everywhere uses TranslationTextComponent

KnightMiner commented on src/main/java/knightminer/inspirations/library/Util.java:
```diff
@@ -43,21 +50,26 @@ public static ResourceLocation getResource(String res) {
 	}
 
+	@Deprecated
 	public static boolean canTranslate(String key) {
```
See [here](https://github.com/SlimeKnights/Mantle/blob/1.14/src/main/java/slimeknights/mantle/item/TooltipItem.java#L32), though it might be client side only. Can move this method to client utils if so (assuming I am actually using this)

KnightMiner commented on src/main/java/knightminer/inspirations/library/Util.java:
```diff
@@ -66,43 +78,57 @@ public static Logger getLogger(String type) {
+	// Using a Stick makes sure it won't be damaged.
+	private static ItemStack silkTouchItem = new ItemStack(Items.STICK);
+	static {
+		silkTouchItem.addEnchantment(Enchantments.SILK_TOUCH, 1);
```
Cleaver hack :)

KnightMiner commented on src/main/java/knightminer/inspirations/library/Util.java:
```diff
@@ -66,43 +78,57 @@ public static Logger getLogger(String type) {
 	 * @return  ItemStack for the state, or ItemStack.EMPTY if a valid item cannot be found
 	 */
-	public static ItemStack getStackFromState(@Nullable IBlockState state) {
+	public static ItemStack getStackFromState(ServerWorld world, @Nullable BlockState state) {
```
Honestly not sure this needs to be so complicated anymore, I only used `getSilkTouchDrop` before as `getMetaFromState` was not reliable for getting a item stack metadata, but now that metadata is gone I can just replace all calls to this method with `new ItemStack(state.getBlock())`

KnightMiner commented on src/main/java/knightminer/inspirations/library/client/ClientUtil.java:
```diff
@@ -36,53 +36,56 @@
 import java.util.Set;
 
-@SideOnly(Side.CLIENT)
+@OnlyIn(Dist.CLIENT)
```
Lex says we are not supposed to use this outside vanilla methods, so might as well follow good practice to start 1.14. This class should be safe as long as we just don't call it anywhere in serverside code

KnightMiner commented on src/main/java/knightminer/inspirations/library/client/ClientUtil.java:
```diff
@@ -36,53 +36,56 @@
+	 * @param stack Input stack
+	 * @return Color for the stack
 	 */
 	public static int getStackColor(ItemStack stack) {
```
Change to `getItemColor` to remove the appearance that NBT has an impact?

KnightMiner commented on src/main/java/knightminer/inspirations/library/recipe/ModItemList.java:
```diff
@@ -3,50 +3,82 @@
+	@Nonnull
+    @Override
+    public IIngredientSerializer<? extends Ingredient> getSerializer()
+    {
```
Bracket on newline

KnightMiner commented on src/main/java/knightminer/inspirations/library/recipe/ShapelessNoContainerRecipe.java:
```diff
@@ -1,48 +1,59 @@
-			if (ings.isEmpty()) {
-				throw new JsonParseException("No ingredients for shapeless recipe");
-			}
+	private static class Serializer extends net.minecraftforge.registries.ForgeRegistryEntry<IRecipeSerializer<?>> implements IRecipeSerializer<ShapelessNoContainerRecipe> {
```
Import ForgeRegistryEntry

KnightMiner commented on src/main/java/knightminer/inspirations/plugins/top/TheOneProbePlugin.java:
```diff
@@ -1,17 +1,17 @@
-		FMLInterModComms.sendFunctionMessage(modid, "getTheOneProbe", "knightminer.inspirations.plugins.top.TheOneProbeRegistrar");
+	@SubscribeEvent
+	public void preInit(InterModEnqueueEvent event) {
+		InterModComms.sendTo(modid, "getTheOneProbe", TheOneProbeRegistrar::new);
```
Did you verify this works when TOP is not installed? (diddo to the WAILA one) I recall always needing at least one static class layer to prevent class loading issues.

KnightMiner commented on src/main/java/knightminer/inspirations/shared/InspirationsShared.java:
```diff
@@ -1,131 +1,130 @@
-	public static ItemStack silverfishPowder;
-	public static ItemStack witherBone;
-	public static ItemStack stoneRod;
+	public static Item lock;
```
Move all of these to their relevant module, they were only in shared since the material item was in shared

KnightMiner commented on src/main/java/knightminer/inspirations/shared/client/TextureModel.java:
```diff
@@ -40,30 +45,30 @@ public TextureModel(IBakedModel originalModel, IModel model, VertexFormat format
-		return retextured.bake(retextured.getDefaultState(), format, ModelLoader.defaultTextureGetter());
+		return model.retexture(textures).bake(
+				SharedClientProxy.modelLoader,
+				(x) -> Minecraft.getInstance().getTextureMap().getSprite((ResourceLocation) x),
```
What type is x here? Does it need an instanceof check, or can we be certain its always resource location?

KnightMiner commented on src/main/java/knightminer/inspirations/tools/InspirationsTools.java:
```diff
@@ -76,172 +76,191 @@
 		}
+
+//		if(Config.shieldEnchantmentTable) {
+//			register(r, new ItemEnchantableShield(), new ResourceLocation("shield"));
```
Are registry subs not working yet?

TeamSpen210 replied:
They are, must have forgotten to enable and test this one.

KnightMiner commented on src/main/java/knightminer/inspirations/tools/client/RenderModArrow.java:
```diff
@@ -1,20 +1,22 @@
-public class RenderModArrow extends RenderArrow<EntityModArrow> {
+import javax.annotation.Nonnull;
+
+public class RenderModArrow extends ArrowRenderer<RedstoneArrow> {
```
Since I have yet to add a second arrow type, might make sense to finally rename this to ChargedArrowRenderer. Especially since if I add more arrows they will likely be their own entities

KnightMiner commented on src/main/java/knightminer/inspirations/tools/item/ItemWaypointCompass.java:
```diff
@@ -1,79 +1,110 @@
-    this.addPropertyOverride(new ResourceLocation("angle"), new WaypointCompassGetter());
+  private DyeColor color;
+
+  public ItemWaypointCompass(DyeColor color) {
```
Rather than passing in DyeColor, I would pass in two color integers for the base color and the needle color. Lets us get rid of `getNeedleColor` below

KnightMiner commented on src/main/java/knightminer/inspirations/tweaks/InspirationsTweaks.java:
```diff
@@ -1,282 +1,177 @@
-			PotionHelper.addMix(decay, glowstone, strongDecay);
-		}
+		// TODO: Forge needs to implement the ability to inject these into the loot tables.
+		// See MinecraftForge#
```
Should just be a loot table now, though honestly I'd just ditch adding them to grass drops (assuming I don't scrap the whole feature)

KnightMiner commented on src/main/java/knightminer/inspirations/tweaks/TweaksEvents.java:
```diff
@@ -60,9 +70,40 @@ public static void unsaddlePig(EntityInteract event) {
 		}
 	}
 
+	// TODO: Replace this with loot table JSONS when Forge implements a system to add pools to existing tables.
```
See [here for how to do this](https://github.com/elucent/SimplyTea/blob/1.14/src/main/java/knightminer/simplytea/core/Events.java#L67-L78). Ignore the TODO comments I forgot to remove when it was fixed

KnightMiner commented on src/main/java/knightminer/inspirations/utility/InspirationsUtility.java:
```diff
@@ -1,156 +1,167 @@
-	public static Block carpetedPressurePlate2;
+	public static Block bricksButton;
+	public static Block netherBricksButton;
+	public static Block[] carpetedTrapdoors = new Block[16];
```
I'd make these consistent with block lists used elsewhere. Current you are using maps, though I'd like to use List if possible.

KnightMiner commented on src/main/resources/META-INF/mods.toml:
```diff
@@ -0,0 +1,64 @@
+    versionRange="[1.4,]"
+    ordering="AFTER"
+    side="BOTH"
+[[dependencies.inspirations]]
```
TAN is RIP, so you can remove this

KnightMiner commented on src/main/java/knightminer/inspirations/plugins/tan/ToughAsNailsPlugin.java:
```diff
@@ -1,21 +1,20 @@
 package knightminer.inspirations.plugins.tan;
```
Can be removed as TAN is RIP

TeamSpen210 committed:
Update WALIA/HYWLA plugin to 1.14

TeamSpen210 committed:
Update some recipes to 1.14

TeamSpen210 committed:
Update squid milking to 1.14

TeamSpen210 committed:
Update Anvil to 1.14

(Untested)

KnightMiner reviewed (changes requested):
Overall looks good. Biggest comment is class naming is a bit inconsistent between 1.12 and 1.14 naming. I'd like to switch to 1.14 naming everywhenre (ThingBlock rather than BlockThing), though that might make the commits harder to read. Then again, its already changing 150 something Java files so its not like its easy to read as it is.

I see a few WIP things still, but thats mostly the stuff I told you to leave for later, so after making the requested changes this should be nearly ready for merge.

I'll leave comments on the gameplay side of things in a bit, wanted to get these code comments out first.

TeamSpen210 committed:
Update build.gradle logic correctly for 1.14

TeamSpen210 committed:
Pass in body/needle colors to Waypoint Compass constructor

This simplifies the tinting logic considerably.

TeamSpen210 committed:
Make dying Waypoint Compass config disable colored compasses

TeamSpen210 committed:
Remove unused constants from Inspirations class

TeamSpen210 committed:
Remove Tough As Nails compatibility

This mod isn't being ported.

TeamSpen210 committed:
Remove unused subscriber annotation

TeamSpen210 committed:
Check the casts here

TeamSpen210 committed:
Rewrite bookshelf collisions logic to be easier to understand

TeamSpen210 committed:
Do networking of Redstone Arrow properly, so it spawns clientside

TeamSpen210 committed:
Consolidate HidableItem logic into a single method

This ensures the code can be changed in a central location.

TeamSpen210 committed:
Reformat some code

KnightMiner commented on src/main/java/knightminer/inspirations/common/IHidable.java:
```diff
@@ -0,0 +1,25 @@
+ * Used for blocks/items that are disableable in the config.
+ */
+public interface IHidable {
+    boolean isEnabled();
```
Should add javadocs for these method rather than inline comments, even if its just
```
/** Some description */
```

KnightMiner commented on src/main/java/knightminer/inspirations/common/block/HidableBlock.java:
```diff
@@ -0,0 +1,30 @@
+        this.enabled = isEnabled;
+    }
+
+    public boolean isEnabled() {
```
Missing `@Override` for interface methods, ideally I always want either `@Override` or javadocs for functions

KnightMiner commented on src/main/java/knightminer/inspirations/common/item/HidableBlockItem.java:
```diff
@@ -0,0 +1,38 @@
+import javax.annotation.Nonnull;
+import java.util.function.Supplier;
+
+public class HidableBlockItem extends BlockItem implements IHidable {
```
Isn't this redundant? If I remember, BlockItem already redirects to the Block's `fillItemGroup` method, so you are just telling it not to show twice

TeamSpen210 replied:
In that sense yes, but I'm wanting every item and block to implement IHidable so we can then hide/show them in JEI and the like.

KnightMiner replied:
In that case, maybe keep the isHidden implementation and skip the fillItemGroup call as it will be handled by the block later on?

KnightMiner commented on src/main/java/knightminer/inspirations/tools/InspirationsTools.java:
```diff
@@ -76,172 +77,213 @@
-	public static ToolMaterial bone;
-	public static ToolMaterial blaze;
-	public static ToolMaterial wither;
+	public static IItemTier bone = new IItemTier() {
```
Assuming I keep the crooks (which I am leaning largely against at this point), I'd rather make a class that implements this interface and do this through the constructor, implementing a whole class inline is awkward.

KnightMiner commented on src/main/java/knightminer/inspirations/utility/block/BlockPipe.java:
```diff
@@ -1,292 +1,266 @@
+
+		// We only need to check the one side that updated.
+		return state
+				.with(HOPPER, offsetState.getBlock() instanceof HopperBlock &&
```
Something feels off about this, the hopper state should only be modified if our facing is the same as the side being updated, its only relevant for the side the pipe looks into.

TeamSpen210 committed:
Re-add ability to sink into mulch blocks

TeamSpen210 committed:
Separate out the different mulch blocks entirely

Get rid of the MulchColor enum, and store each in their own fields directly

TeamSpen210 committed:
Fix rope breaking effects

TeamSpen210 committed:
Add javadocs/@override to Hidable base classes

TeamSpen210 committed:
Entity IDs have been removed.

TeamSpen210 committed:
Only update Pipe hopper state on the side it's attached to

TeamSpen210 committed:
Remove Crooks

Additionally Wither Bone and Stone Rod are then useless.

TeamSpen210 committed:
Switch colored books and carpets to be stored in arrays.

TeamSpen210 committed:
Split Chain variant out into separate class

TeamSpen210 committed:
Use correct material colors for a number of blocks

TeamSpen210 committed:
Tweak field name

TeamSpen210 committed:
Tweak possible values for SpongeEmptyCauldron to be more meaningful

Backwards compat with 1.12 configs isn't important or possible.

TeamSpen210 committed:
Use method references for inspirations:config, instead of a lambda->switch

This extra method is no longer useful, since we don't need to remember
the property string.

TeamSpen210 committed:
Make inspirations:pulse function again

TeamSpen210 committed:
Move much of PulseBase's code into IRegisterUtil

TeamSpen210 committed:
Protect access to clientside MC in packet code

TeamSpen210 committed:
Prevent InsiprationsRegistry from needing to import Config

TeamSpen210 committed:
Eliminate use of wildcard imports.

TeamSpen210 committed:
Don't duplicate this constant

KnightMiner commented on src/main/java/knightminer/inspirations/tools/ToolsClientProxy.java:
```diff
@@ -2,67 +2,40 @@
+			int needleColor = compass.getNeedleColor();
+			int bodyColor = compass.getBodyColor();
+
+			itemColors.register((stack, tintIndex) -> {
```
Do you think this would be cleaner if the compass had a method of the signature `int itemColors(ItemStack,int)` so we can just call `itemColors.register(compass::itemColors, compass)`? Alternatively, if it implemented `IItemColors` so we could call `itemColors.register(compass, compass)`.

The current way is fine, so its just if you think it might be better like that.

TeamSpen210 replied:
Implementing `IItemColors` is a much better idea, yeah.

KnightMiner commented on src/main/resources/assets/inspirations/blockstates/black_carpeted_pressure_plate.json:
```diff
@@ -0,0 +1,11 @@
+    },
+    "powered=true": {
+      "model": "inspirations:block/carpet/plate/black_pressed",
+      "y": 180
```
Why the rotation? This seems weird when you can just have the original model rotated the proper way

TeamSpen210 replied:
I am not sure where this came from.

TeamSpen210 committed:
Update Mantle and HWYLA versions

TeamSpen210 committed:
Carpeted Pressure Plates do not have item versions

TeamSpen210 committed:
Add item models for bookshelf blocks

TeamSpen210 committed:
Leave horizontal carpeted trapdoors unrotated

This is necessary to make them blend in with regular carpet

TeamSpen210 committed:
Don't rotate pressed carpeted pressure plates

TeamSpen210 committed:
Make Waypoint Compasses implement IItemColor directly

TeamSpen210 committed:
Remove the pre/post/init methods

These are obsoleted by the various new events.

TeamSpen210 committed:
Fix carpeted pressure plate pick-block
This needs to be dynamically looked up later on.

TeamSpen210 committed:
Move lock, key, heartbeet and silverfish powder code to their pulses.

TeamSpen210 committed:
Use the correct reference to these items here.

TeamSpen210 committed:
Implement texturing block drops as a loot table function

TeamSpen210 committed:
Make inventory blocks preserve any custom name when dropped

TeamSpen210 committed:
Update Forge
Fluids now available!

TeamSpen210 committed:
Reimplement config recipe conditions for the Forge changes

TeamSpen210 committed:
Correctly fill the ItemGroups with all textureable block types.

KnightMiner commented on src/main/java/knightminer/inspirations/utility/block/BlockCarpetedPressurePlate.java:
```diff
@@ -36,8 +41,19 @@ public VoxelShape getShape(@Nonnull BlockState state, @Nonnull IBlockReader worl
+		if (pickItem == Items.AIR) {
+			pickItem = ForgeRegistries.ITEMS.getValue(new ResourceLocation(color.getTranslationKey() + "_carpets"));
+		}
+		if(pickItem == Items.AIR) {
```
Nest this in the previous `if`, so we don't have to run the check twice when we already have the carpet

TeamSpen210 committed:
Make sure to call super.onReplaced()

Without this, the TE isn't removed on destruction.

TeamSpen210 committed:
Merge 1.12 code into 1.14, discarding redundant 1.12 code

TeamSpen210 committed:
Update Cauldron recipe code for 1.14 Forge changes

TeamSpen210 committed:
Rearrange this if statement to prevent checking the item twice.

TeamSpen210 committed:
Apply a dummy translation key to carpeted pressure plates

TeamSpen210 committed:
Use utility method to check if BlockCollector should be added to the item group

TeamSpen210 committed:
Update Enlightened Bush to 1.14

TeamSpen210 committed:
Update getStackTexture() to return ItemStack.EMPTY if the stack isn't present.

TeamSpen210 committed:
Special-case Enlightened Bush so that it can be in the tag of
valid texture stacks for itself

Don't actually allow it to be recursively crafted.

TeamSpen210 committed:
Correct new import location for Fluids

TeamSpen210 committed:
Implement a more accurate collision shape for Carpeted Pressure plates

Does mean the plate itself is more obvious, it's not really hidden though.

TeamSpen210 committed:
Rename RenderModArrow to RedstoneArrowRenderer

TeamSpen210 committed:
Fix formatting

TeamSpen210 committed:
Tweak this lambda to make it clear that this is a safe cast.

TeamSpen210 committed:
Make ConfigEnabled and PulseLoaded loot table conditions as well
as recipe conditions

TeamSpen210 committed:
Use loot tables for cave spider and skeleton extra drops

TeamSpen210 committed:
The entity equipment slots may not be present, null-check them.

TeamSpen210 committed:
Change this function name to make it obvious it ignores stack info

TeamSpen210 committed:
OnlyIn should only be used in vanilla code

TeamSpen210 committed:
Move loot table append code back to PulseBase

Missed the old function being here...

TeamSpen210 committed:
Fluid sounds are in the attributes now.

TeamSpen210 committed:
Make Enchantable Shields fully functional

TeamSpen210 committed:
Rewrite small flower dropping to use loot tables

TeamSpen210 committed:
Remove "Raw" suffix from hidden config fields

TeamSpen210 committed:
Make BubbleParticle constructor public
The cauldron boiling particle needs this.

KnightMiner commented on src/main/java/knightminer/inspirations/building/BuildingClientProxy.java:
```diff
@@ -173,10 +185,24 @@ public void onModelBake(ModelBakeEvent event) {
+	}
+
+	private static void replaceBushModel(ModelBakeEvent event, BlockEnlightenedBush bush) {
+		ResourceLocation bushLoc = Objects.requireNonNull(bush.getRegistryName());
```
This function seems pretty generic, any reason it is not just pulled into a variant of replaceTextureModel? Something that replaces both a block and an item model based on the registry name

TeamSpen210 replied:
Not really no, I should do so.

KnightMiner commented on src/main/java/knightminer/inspirations/building/BuildingClientProxy.java:
```diff
@@ -173,10 +185,24 @@ public void onModelBake(ModelBakeEvent event) {
+		ResourceLocation itemLoc = new ModelResourceLocation(bushLoc, "inventory");
+		IModel model = ModelLoaderRegistry.getModelOrLogError(itemLoc,"Error loading item model for " + itemLoc);
+		IBakedModel standard = event.getModelRegistry().get(itemLoc);
+		IBakedModel itemModel = new TextureModel(standard, model, DefaultVertexFormats.ITEM, "texture", true);
```
Just make this a parameter on `replaceTextureModel` for `DefaultVertexFormats.ITEM` (boolean or vertex format) instead of copying all the code over

TeamSpen210 committed:
Rearrange replaceTexturedModel code to handle the variants on the logic

TeamSpen210 committed:
Switch Utils code to keep ITextComponent instead of directly translating

KnightMiner commented on src/main/java/knightminer/inspirations/common/ClientProxy.java:
```diff
@@ -54,11 +57,22 @@ protected static void registerItemColors(ItemColors itemColors, IItemColor handl
+		);
+	}
+
+	protected static void replaceBothTexturedModels(ModelBakeEvent event, ResourceLocation loc, String key) {
```
I would suggest an overload for this which takes a block instead of a resource location as the second parameter, since 90% of usages just calll `getRegistryName`
